cli/pack/common.go

```diff
@@ -108,33 +111,59 @@ func detectVersion(ctx *context.Ctx) error {
 	return nil
 }
 
-func getPackageFullname(ctx *context.Ctx) string {
-	ext, found := extByType[ctx.Pack.Type]
-	if !found {
-		panic(project.InternalError("Unknown type: %s", ctx.Pack.Type))
+func getPackageSuffix(ctx *context.Ctx) string {
+	var suffix string
+	if ctx.Pack.Suffix != "" {
+		suffix = fmt.Sprintf("-%s", ctx.Pack.Suffix)
 	}
 
-	packageFullname := fmt.Sprintf(
-		"%s-%s",
+	return suffix
+}
+
+func getRpmPackageNameBody(ctx *context.Ctx) string {
+	return fmt.Sprintf(
+		"%s-%s%s.%s",
 		ctx.Project.Name,
 		ctx.Pack.VersionRelease,
+		getPackageSuffix(ctx),
+		runtime.GOARCH,
 	)
+}
 
-	if ctx.Pack.Suffix != "" {
-		packageFullname = fmt.Sprintf(
-			"%s-%s",
-			packageFullname,
-			ctx.Pack.Suffix,
-		)
+func getDebPackageNameBody(ctx *context.Ctx) string {
+	return fmt.Sprintf(
+		"%s_%s-%s%s_%s",
+		ctx.Project.Name,
+		ctx.Pack.VersionRelease,
+		debianVersion,
+		getPackageSuffix(ctx),
+		runtime.GOARCH,
+	)
+}
+
+// TGZ and Docker
+func getCommonPackageNameBody(ctx *context.Ctx) string {
+	return fmt.Sprintf("%s-%s%s", ctx.Project.Name, ctx.Pack.VersionRelease, getPackageSuffix(ctx))
+}
+
+func getPackageFullname(ctx *context.Ctx) string {
+	ext, found := extByType[ctx.Pack.Type]
+	if !found {
+		panic(project.InternalError("Unknown type: %s", ctx.Pack.Type))
 	}
 
-	packageFullname = fmt.Sprintf(
-		"%s.%s",
-		packageFullname,
-		ext,
-	)
+	var packageFullname string
+
+	switch ctx.Pack.Type {
+	case RpmType:
+		packageFullname = getRpmPackageNameBody(ctx)
+	case DebType:
+		packageFullname = getDebPackageNameBody(ctx)
```

Comment:
Why not simple `get{Rpm,Deb}PackageFullname`? In fact, `ext` depends on pack type.  The default case is used for TGZ only.